src/Nethermind/Nethermind.Synchronization/SnapSync/SnapProviderHelper.cs

```diff
@@ -29,9 +29,20 @@ public static (AddRangeResult result, bool moreChildrenToRight, List<PathWithAcc
             IReadOnlyList<byte[]> proofs = null
         )
         {
-            // TODO: Check the accounts boundaries and sorting
             if (accounts.Count == 0)
                 throw new ArgumentException("Cannot be empty.", nameof(accounts));
+            // Validate sorting and boundary range
+            for (int i = 1; i < accounts.Count; i++)
+            {
+                if (accounts[i - 1].Path >= accounts[i].Path)
+                    return (AddRangeResult.InvalidOrder, true, null, null);
+            }
+
+            for (int i = 0; i < accounts.Count; i++)
```

Comment:
Need to check how it actually affects the sync process. It seems to be common to receive a node after the limit path and it is handled in the code a few lines below. By rejecting such responses, we might be stalling the sync process.
https://github.com/NethermindEth/nethermind/blob/a0822b048787ca7bdfa192dcac2a906857c58f93/src/Nethermind/Nethermind.Synchronization/SnapSync/SnapProviderHelper.cs#L56
https://github.com/NethermindEth/nethermind/blob/a0822b048787ca7bdfa192dcac2a906857c58f93/src/Nethermind/Nethermind.Synchronization/SnapSync/SnapProviderHelper.cs#L83